floodmodeller_api/toolbox/model_build/structure_log/structure_log.py

```diff
@@ -19,237 +53,445 @@ def _add_fields(self):
+            "bore_area": structure.bore_area,
+        }
+        dimensions["height"] = structure.soffit - structure.invert
         if structure.shape == "RECTANGLE":
-            height = structure.soffit - structure.invert
-            width = structure.bore_area / height
-            dimensions = f"h: {height:.2f} x w: {width:.2f}"
+            dimensions["width"] = structure.bore_area / dimensions["height"]
+            dimensions["bore_area"] = structure.bore_area
         elif structure.shape == "CIRCULAR":
-            diameter = structure.soffit - structure.invert
-            dimensions = f"dia: {diameter:.2f}"
-        return dimensions
+            dimensions["width"] = dimensions["height"]
+            dimensions["bore_area"] = (dimensions["height"] ** 2) * (
+                3.1415 * 0.25
+            )  # calcuate bore area from given diameter
+        return {"dimensions": dimensions}
 
     def _spill_data(self, structure):
-        elevation = min(structure.data.Y.tolist())
         x_list = structure.data.X.tolist()
-        width = max(x_list) - min(x_list)
-        dimensions = f"Elevation: {elevation:.2f} x w: {width:.2f}"
-        weir_coefficient = structure.weir_coefficient
-        return [dimensions, weir_coefficient]
+        y_list = structure.data.Y.tolist()
+        dimensions = {
+            "invert": min(y_list),
+            "width": max(x_list) - min(x_list),
+            "weir_coefficient": structure.weir_coefficient,
+            "section_data": {"x": x_list, "y": y_list},
+        }
+        return {"dimensions": dimensions}
 
     def _bridge_data(self, structure):
-        all_mannings = structure.section_data["Mannings n"].tolist()
-        mannings_set = {min(all_mannings), max(all_mannings)}
-        if len(mannings_set) == 1:
-            friction = f"Mannings: {mannings_set.pop()}"
-        else:
-            friction = f"Mannings: [min: {mannings_set.pop()}, max: {mannings_set.pop()}]"
-        height = structure.opening_data.values[0][3] - min(structure.section_data.Y.tolist())
-        width = structure.opening_data.values[0][1] - structure.opening_data.values[0][0]
-        dimensions = f"h: {height:.2f} x w: {width:.2f}"
-        return [friction, dimensions]
+        section_df = structure.section_data
+        all_friction = section_df["Mannings n"].tolist()
+        friction_set = sorted(set(all_friction))
+        friction = {"friction_set": friction_set, "all_friction": all_friction}
+
+        orifice_data = extract_attrs(
+            structure,
+            {
+                "orifice_flow",
+                "orifice_lower_transition_dist",
+                "orifice_upper_transition_dist",
+                "orifice_discharge_coefficient",
+            },
+        )
+        orifice_data["transition_width"] = (
+            structure.orifice_upper_transition_dist + structure.orifice_lower_transition_dist
+        )
+
+        opening_data = []
+        if (
```

Comment:
I think the mix of comments and the black formatter is making this look a bit weird - I would move the comments onto their own line so black doesn't try to split all these if statements & loops into multiple lines, if that makes sense?

Reply:
addressed, hopefully black doesnt butcher it again

Reply:
I don't think this is addressed, unless you haven't pushed this change?

Reply:
should be fixed now